### PR TITLE
Add query frontend to the quickstart script

### LINKED ISSUE #3304: Add query frontend component to the quickstart script
# Feature request

It would be nice to integrate query frontend to the https://github.com/thanos-io/thanos/blob/master/scripts/quickstart.sh so that we can easily test the whole stack.

AC:
1. In-memory result cache config should be supported
2. Would be nice to configure Jaeger for query frontend

### PR DESCRIPTION
**Pull request by vears91**
Closes #3304

Signed-off-by: Victor Araujo <vear91@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: scripts/quickstart.sh [hunk 1]
```diff
@@ -262,6 +286,16 @@ for i in $(seq 0 1); do
     --tracing.config="${QUERIER_JAEGER_CONFIG}" \
     --query.replica-label receive_replica \
     ${STORES} &
+
+  sleep 0.5
+
+  ${THANOS_EXECUTABLE} query-frontend \
```
Comment (bwplotka) [inline scripts/quickstart.sh:292]: This is not how it suppose to work, unfortunately. Those replicas have to be separate for better horizontal scalability. This means there should be a load balancing in between - ideally round robin load balancer for better distribution.

We can add some nginx for it, but ideally, we could work towards this: https://github.com/thanos-io/thanos/issues/3373

Let me know if this makes sense :hugs:

**Review by bwplotka — CHANGES_REQUESTED** [review]
Nice thanks! 

It looks amazing, except one thing that might be blocked first by https://github.com/thanos-io/thanos/issues/3373

File: scripts/quickstart.sh [hunk 2]
```diff
@@ -11,6 +11,9 @@ MC_EXECUTABLE=${MC_EXECUTABLE:-"mc"}
 PROMETHEUS_EXECUTABLE=${PROMETHEUS_EXECUTABLE:-"prometheus"}
 THANOS_EXECUTABLE=${THANOS_EXECUTABLE:-"thanos"}
 S3_ENDPOINT=""
+QUERY_FRONTEND_CACHE_MAX_SIZE=${QUERY_FRONTEND_CACHE_MAX_SIZE:-""}
+QUERY_FRONTEND_CACHE_MAX_SIZE_ITEMS=${QUERY_FRONTEND_CACHE_MAX_SIZE_ITEMS:-0}
+QUERY_FRONTEND_CACHE_VALIDITY=${QUERY_FRONTEND_CACHE_VALIDITY:-"0s"}
```
Comment (yeya24) [inline scripts/quickstart.sh:16]: Let's use `6h` here

File: scripts/quickstart.sh [hunk 3]
```diff
@@ -11,6 +11,9 @@ MC_EXECUTABLE=${MC_EXECUTABLE:-"mc"}
 PROMETHEUS_EXECUTABLE=${PROMETHEUS_EXECUTABLE:-"prometheus"}
 THANOS_EXECUTABLE=${THANOS_EXECUTABLE:-"thanos"}
 S3_ENDPOINT=""
+QUERY_FRONTEND_CACHE_MAX_SIZE=${QUERY_FRONTEND_CACHE_MAX_SIZE:-""}
+QUERY_FRONTEND_CACHE_MAX_SIZE_ITEMS=${QUERY_FRONTEND_CACHE_MAX_SIZE_ITEMS:-0}
```
Comment (yeya24) [inline scripts/quickstart.sh:15]: Let's set this to 2048

File: scripts/quickstart.sh [hunk 4]
```diff
@@ -11,6 +11,9 @@ MC_EXECUTABLE=${MC_EXECUTABLE:-"mc"}
 PROMETHEUS_EXECUTABLE=${PROMETHEUS_EXECUTABLE:-"prometheus"}
 THANOS_EXECUTABLE=${THANOS_EXECUTABLE:-"thanos"}
 S3_ENDPOINT=""
+QUERY_FRONTEND_CACHE_MAX_SIZE=${QUERY_FRONTEND_CACHE_MAX_SIZE:-""}
```
Comment (yeya24) [inline scripts/quickstart.sh:14]: Let's set this to `"0"`

**Comment by kakkoyun** [comment]
@vears91 Do you want to continue working on it? I can open it and give it another round of review. Sorry if we dropped the ball on this one.

**Comment by vears91** [comment]
> @vears91 Do you want to continue working on it? I can open it and give it another round of review. Sorry if we dropped the ball on this one.

Hi. I would like to, is https://github.com/thanos-io/thanos/issues/3373 still blocking this?

**Pull request closed without merging**